text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,811 @@
+- **Resource Provider (new)**: A component responsible for taking Resource and Entity detectors and doing the following:
+  - Constructing a Resource for the SDK from detectors.
+  - Dealing with conflicts between detectors.
+  - Providing SDK-internal access to detected Resources for reporting via Log signal on configured LogProviders.
+  - *(new) Managing Entity changes during SDK lifetime, specifically dealing with entities that have lifetimes shorter than the SDK*
+
+#### Resource Provider
+
+The SDK Resource Provider is responsible for running all configured Resource and Entity Detectors.  There will be some (user-controlled, otel default) priority order to these.
+
+- The Resource Provider will detect conflicts in Entity of the same type being discovered and choose one to use.
+- When using Entity Detectors and Resource detectors together, the following merge rules will be used:
+  - Entity merging will occur first resulting in an "Entity Merged" Resource (See [algorithm here](#entity-merging-and-resource)).
+  - Resource detectors otherwise follow existing merge semantics.
+    - The Specification merge rules will be updated to account for violations prevalent in ALL implementation of resource detection.
+    - Specifically: This means the [rules around merging Resource across schema-url will be dropped](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#merge).  Instead only conflicting attributes will be dropped.
+    - SchemaURL on Resource will be deprecated with entity-specific schema-url replacing it. SDKs will only fill out SchemaURL on Resource when SchemaURL matches across all entities discovered. Additionally, only existing stable resource attributes can be used in Resource SchemaURL in stable OpenTelemetry components (Specifially `service.*` and `sdk.*` are the only stabilized resource convnetions). Given prevalent concerns of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal, and existing usage was within the "experimental" phase of semantic conventions.
+  - An OOTB ["Env Variable Entity Detector"](#environment-variable-detector) will be specified and provided vs. requiring SDK wide ENV variables for resource detection.
+- *Additionally, Resource Provider would be responsible for understanding Entity lifecycle events, for Entities whose lifetimes do not match or exceed the SDK's own lifetime (e.g. browser session).*
+
+#### Entity Detector
+
+The Entity detector in the SDK is responsible for detecting possible entities that could identify the SDK (called "associated entities"). For Example, if the SDK is running in a kubernetes pod, it may provide an Entity for that pod. SDK Entity Detectors are only required to provide identifying attributes, but may provide descriptive attributes to ensure combined Resource contains similar attributes as today's SDK.
```

Comment:
From my practical experience, I doubt that it *can* be more specific and meaningful.

Reply:
This is use case based.  There are some descriptive attributes that are valuable to group/slice by in resource used today (see the list of resource attributes listed below which are not identifying).

This is aimed to not be too prescriptive so we can preserve existing resource behavior.  I think for semantic conventions we'll want to provide a "bar" on when to do this, but for general specification and users, we should leave it open.

Reply:
Would it make sense to further distinguish 'required' attributes per entity type which are then always included in descriptive attributes? Otherwise to determine which attributes to expect one would have to 'read the memo' per SDK?

Reply:
Yes - but this would be done in semantic conventions, similar to what we do for metrics/spans/logs: see [http spans](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-client) for example.

I.e. I think the "requirement level" is ok to be defined separate from the data itself.  The notion of identity + descriptive should be encoded in the transmitted data - i.e. we assume all required data is already on the entity when we get it.

Reply:
I doubt that requirements levels have any practical relevance in the sense of a software contract. Equally, it is not plausible that a descriptive attribute is required. "desired" would be a better word. But that's a digression and the discussion should be resolved here.